### ISSUE TITLE
Label at the end of a block is not a whitespace (nor comment)

### ISSUE THREAD

**Issue by olibre** [bombsimon/wsl#92]
Simplified version of my code:

```go
for _, x := range a {
	if x == y {
		goto SKIP // do not append
	}

	clean = append(clean, x)
SKIP:
}       // block should not end with a whitespace (or comment)
```

I think `wsl` considers the label `SKIP` as a whitespace (or comment)...

**Comment by bombsimon** [comment]
I don't know why you would do this in golang though, it seems lika a pattern from BASIC och batch scripting. Maybe that says more about me but I thought I've seen my shared amount of Go code and this pattern never occurred to me. In your example I would just use `break` to exit the loop. If I wanted to break out of nested loops I would label the outer most loop and break that one like so:

```go
OUTER:
    for range a {
        for _, x := range b {
            if x == y {
                break OUTER
            }
        }
    }
```

If I had code that wasn't related to this loop that I wanted to skip i would use `return`. If I had code after this loop that I wanted to scope and more code after that that I wanted to execute I would break this function into several functions since that sounds like it's doing way to much.

I tried to Google and see when a good use case for `goto` in golang would be and I found some references to stdlib. Those however is never to go to the end of a scope and especially not to the end of a function.
  
So basically this is not intended but since I want this linter to enforce good practices not only by adding whitespaces but to think of what's a good way to tackle an issue and improve readability I don't see this as something that I do want to support. Ending a scope with a label just to be able to `goto` it is not something I want to encourage.

Obviously this is my personal opinion so feel free to argue why you think this is a good pattern and what the benefits over the other control flows such as `break`, `countinue` and `return` this would have.

**Comment by olibre** [comment]
Thank you for your excellent advise :+1: 
My new code:

```
func removeDuplicates(a, b []string) []string {
	clean := []string{}

OUTER:
	for _, x := range a {
		for _, y := range b {
			if x == y {
				goto OUTER // do not append
			}
		}

		clean = append(clean, x)
	}

	return clean
}
```

Note: Coding in Go is new for me :wink:

**Comment by olibre** [comment]
Sorry for the four PR: I was thinking I was editing my own forked of your Git repo!

**Comment by Oppodelldog** [comment]
I agree with @olibre, a jump label is not a whitespace and should not be treaten as if it was.
@bombsimon do you know what is the intention of doing so? Isn't "good coding practise" out of scope for a whitespace linter?

**Comment by bombsimon** [comment]
@Oppodelldog yep, you’re right. This should not be a part of wsl so it should be fixed, that’s why I left the issue open. This lintet should not warn about this! PRs welcome! :)

**Comment by Oppodelldog** [comment]
@bombsimon I'll give it a try in hacktober. ;)

**Comment by bombsimon** [comment]
Merged in https://github.com/golangci/golangci-lint/pull/1750, will be available in next release.
